src/Microsoft.Azure.SignalR.Management/RestHubLifetimeManager.cs

```diff
@@ -196,6 +198,26 @@ public Task UserRemoveFromAllGroupsAsync(string userId, CancellationToken cancel
             return _restClient.SendAsync(api, HttpMethod.Delete, _productInfo, cancellationToken: cancellationToken);
         }
 
+        public async Task<bool> IsUserInGroup(string userId, string groupName, CancellationToken cancellationToken = default)
+        {
+            var isUserInGroup = false;
+            var api = _restApiProvider.GetUserGroupManagementEndpoint(_appName, _hubName, userId, groupName);
+            await _restClient.SendAsync(api, HttpMethod.Get, _productInfo, handleResponse: (request, response) =>
+                {
+                    switch (response.StatusCode)
+                    {
+                        case HttpStatusCode.OK:
+                            isUserInGroup = true;
+                            return Task.CompletedTask;
+                        case HttpStatusCode.NotFound:
+                            return Task.CompletedTask;
+                        default:
+                            return _restClient.ThrowExceptionOnResponseFailureAsync(request, response);
```

Comment:
a bit confused with invoke `_restClient.ThrowExceptionOnResponseFailureAsync` in the callback function.
what about following:
Callback returns `true` for handled, `false` for unhandled.
If callback return false (i.e. unhandled), then `SendAsync` call `ThrowExceptionOnResponseFailureAsync`.

Reply:
How about changing the `handleResponse` to `Func<HttpResponseMessage, Func<HttpResponseMessage>, Task>`? (We don't need `HttpRequestMessage`, since `HttpResponseMessage` already contains an property for it).
```
public async Task<bool> IsUserInGroup(string userId, string groupName, CancellationToken cancellationToken = default)
        {
            var isUserInGroup = false;
            var api = _restApiProvider.GetUserGroupManagementEndpoint(_appName, _hubName, userId, groupName);
            await _restClient.SendAsync(api, HttpMethod.Get, _productInfo, handleResponse: ( response, ensureSuccessStatusCode) =>
                {
                    switch (response.StatusCode)
                    {
                        case HttpStatusCode.OK:
                            isUserInGroup = true;
                            return Task.CompletedTask;
                        case HttpStatusCode.NotFound:
                            return Task.CompletedTask;
                        default:
                            return ensureSuccessStatusCode(response); // change ThrowExceptionOnResponseFailureAsync to ensureSuccessStatusCode, since HttpResponseMessage already use this name, resue it here
                    }
                }, cancellationToken: cancellationToken);
            return isUserInGroup;
        }
```

Reply:
If `handleResponse` returns true/false,
then if `handleResponse` returns true, we need to write the throw exception code in the `handleResponse`, since we won't call `ThrowExceptionOnResponseFailureAsync`.
or if `handleResponse` returns false, we still can't skip some failure status code (e.g. 404 in this case) in `ThrowExceptionOnResponseFailureAsync`.

Reply:
I means:
```cs
public async Task<bool> IsUserInGroup(string userId, string groupName, CancellationToken cancellationToken = default)
{
    var isUserInGroup = false;
    var api = _restApiProvider.GetUserGroupManagementEndpoint(_appName, _hubName, userId, groupName);
    await _restClient.SendAsync(api, HttpMethod.Get, _productInfo, handleResponse: response =>
    {
        switch (response.StatusCode)
        {
            case HttpStatusCode.OK:
                isUserInGroup = true;
                return Task.FromResult(true);
            case HttpStatusCode.NotFound:
                return Task.FromResult(true);
            default:
                return Task.FromResult(false);
        }
    }, cancellationToken: cancellationToken);
    return isUserInGroup;
}
```
An exception should be always throw, when your handler returns `false`.
If you do not pass the handler, the default handler should check status code is 2XX.

Reply:
And it is better to have overloads for handleResponse, one for `Task<bool>`, another for `bool`